src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -12238,6 +12297,9 @@ class CL1
                 // (27,16): error CS0165: Use of unassigned local variable 'x3'
                 //         M2(ref x3);
                 Diagnostic(ErrorCode.ERR_UseDefViolation, "x3").WithArguments("x3").WithLocation(27, 16),
+                // (27,16): warning CS8600: Converting null literal or possible null value to non-nullable type.
```

Comment:
This error message reads like it applies to the input of the `ref`, not the fact that `M2` could assign `null` into `x3` in the body. Perhaps a better error message would be something like `'M2' can assign 'null' into 'x3'`.

Reply:
This is going to be tricky to fix. This should not be a safety warning, but a W-warning. The problem is that we have a single W-warning (8600), thus a single diagnostic message...
The message isn't great, but it is functionally correct, so I'm tempted to leave as-is for now. We could file an issue... Let me know what you think.

---
In reply to: [258170299](https://github.com/dotnet/roslyn/pull/33447#discussion_r258170299) [](ancestors = 258170299)